asset/asset.go

```diff
@@ -726,19 +745,63 @@ func NewScriptKeyBip86(rawKey keychain.KeyDescriptor) ScriptKey {
 
 // DeriveGroupKey derives an asset's group key based on an internal public
 // key descriptor, the original group asset genesis, and the asset's genesis.
-func DeriveGroupKey(genSigner GenesisSigner, rawKey keychain.KeyDescriptor,
-	initialGen Genesis, currentGen *Genesis) (*GroupKey, error) {
+func DeriveGroupKey(genSigner GenesisSigner, genBuilder GenesisTxBuilder,
+	rawKey keychain.KeyDescriptor, initialGen Genesis,
+	newAsset *Asset) (*GroupKey, error) {
```

Comment:
I think we only need the `newAsset` argument as the genesis is now always the same (the one used to derive the key in the first place), and the asset include a `Genesis` instance.

Reply:
Not sure what you mean, `initialGen == newAsset.Genesis` only for the group creation. We also use the full group anchor genesis for an asset type check here instead of just passing the single tweak.